packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -286,7 +291,7 @@ export class PendingTransactionTracker {
 
     if (!rawTx?.length) {
       log('Approving transaction as no raw value');
-      await this.#approveTransaction(txMeta.id);
+      await this.#approveTransaction(txMeta.id, true);
```

Comment:
I'd vote to avoid adding more arguments and complexity to the approve method since it's a fundamental part of the flow.

I think this issue is already handled via the `try catch` in the `#resubmitTransactions` method and the associated `KNOWN_TRANSACTION_ERRORS` constant.

Could it be that an RPC provider is just an error message we don't recognise so we can just update the list?